src/lib/utilities.js

```diff
@@ -44,3 +45,64 @@ export function uuid() {
     s4() + '-' + s4() + s4() + s4();
 }
 
+
+// The following is all for global error handling.
+// We keep the globalError value on top of AppController's 
+// this.state.error because this.setState is asynchronous
+// while this variable will be synchronous and can therefore
+// always be checked
+let globalError = null;
+let appControllerMounted = false;
+
+// These will be the functions with 'this' bound to AppController
+// So they can change the state of it globally
+let boundSetGlobalError, boundResetGlobalError;
+
+// Remember to call this function with the correct 'this' value
+export function setContextForGlobalErrorFunctions() {
+  if (appControllerMounted) {
+    throw new Error("AppController already mounted, context is already set");
+  }
+  // Arrow function automatically binds this
+  boundSetGlobalError = (err) => {
+    this.safeSetState({error: err});
+  }
+  boundResetGlobalError = () => {
+    this.safeSetState({error: null, displayErrorMessage: false});
+  }
+  appControllerMounted = true;
+}
+
+// This function doesn't actually reset the context,
+// but will just be called when appController has been unmounted and
+// the context therefore has become invalid.
+export function resetContextForGlobalErrorFunctions() {
+  appControllerMounted = false;
+  globalError = null;
+}
+
+export function resetGlobalError() {
```

Comment:
I think if you are gonna track globalError, track it only in one place. My recommendation would be to keep it on the state of the error component, and update it always using this.safeSetState